src/main/java/com/uwetrottmann/tmdb2/TmdbInterceptor.java

```diff
@@ -39,10 +70,95 @@ public static Response handleIntercept(Chain chain, String apiKey) throws IOExce
         HttpUrl.Builder urlBuilder = request.url().newBuilder();
         urlBuilder.setEncodedQueryParameter(Tmdb.PARAM_API_KEY, apiKey);
 
+        if (request.url().pathSegments().get(0).equals("account")) {
```

Comment:
I just developed the Solution, but before pushing i state here how it works, and if you have any objection i will change it again.

As you suggested there are 2 methods on the Tmdb Class, accountSession(String,String) and guestSession(void), they store the appropriate information on Tmdb, and beside that, they **DO NOT AUTHENTICATE THE USER**. They can both work simultaneously. On each conflicted method, there's an Optional AuthenticationType field, with the above implementation. It can be omitted, If is defined, the interceptor captures this query parameter, analyzes it, stores it as the activeRequestAuthType, on the Tmdb class,, and removes it from the query string, then processes the request, if it gets 401, authenticator is getting called, by the activeRequestAuthType, determines how he should proceed with the authentication. 
if the AuthenticationType field is omitted, then the interceptor determines the best authentication method to go based on the user's configuration. If user has called accountSession() and guestSession(), it will prioritize the Account Session. If only one has been called, will choose this one, if noone has been called, authenticator will just return null, and the request will fail with 401.